src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -183,11 +171,11 @@ void OTARequestor::OnApplyUpdateResponse(void * context, const ApplyUpdateRespon
         break;
     case OTAApplyUpdateAction::kAwaitNextAction:
         requestorCore->mOtaRequestorDriver->UpdateSuspended(System::Clock::Seconds32(response.delayedActionTime));
-        SetUpdateStateAttribute(OTAUpdateStateEnum::kDelayedOnApply);
+        requestorCore->RecordNewUpdateState(OTAUpdateStateEnum::kDelayedOnApply, OTAChangeReasonEnum::kDelayByProvider);
```

Comment:
Likewise, `RecordNewUpdateState` is a utility function which logs the state transition as well as update the server attribute so all previous explicit calls to `SetUpdateStateAttribute` will be replaced by this function.